docs/src/man/hosting.md

```diff
@@ -328,7 +328,9 @@ These are needed to avoid committing generated content to your repository.
 By default, Documenter pushes documentation to the `gh-pages` branch. If the branch does not
 exist it will be created automatically by [`deploydocs`](@ref). If it does exist then
 Documenter simply adds an additional commit with the built documentation. You should be
-aware that Documenter may overwrite existing content without warning.
+aware that Documenter may overwrite existing content without warning. To properly deploy your site
+make sure that you have `gh-pages branch` selected in your GitHub repository settings under
+GitHub Pages.
```

Comment:
This is a good addition. Perhaps we could have it as a separate paragraph even, as the last one in this section?

And it might be also good to link to this GitHub Help page? https://help.github.com/en/github/working-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site